lib/features/give/pages/bt_scan_page.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_blue_plus/flutter_blue_plus.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/features/give/bloc/bloc.dart';
+import 'package:givt_app/features/give/pages/giving_page.dart';
+import 'package:givt_app/features/give/pages/organization_list_page.dart';
+import 'package:givt_app/injection.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/utils/app_theme.dart';
+
+class BTScanPage extends StatefulWidget {
+  const BTScanPage({super.key});
+
+  @override
+  State<BTScanPage> createState() => _BTScanPageState();
+}
+
+class _BTScanPageState extends State<BTScanPage> {
+  late FlutterBluePlus flutterBlue;
+  bool _isVisible = false;
+
+  @override
+  void initState() {
+    super.initState();
+    flutterBlue = FlutterBluePlus.instance;
+    flutterBlue
+      ..startScan(timeout: const Duration(seconds: 30)).timeout(
+        const Duration(seconds: 10),
+        onTimeout: () {
+          ///TODO event for timeout to fetch for previous scan donations
+          //context.read<GiveBloc>().add(const GiveBTScanTimeout());
+          setState(() {
+            _isVisible = !_isVisible;
+          });
+        },
+      )
+      ..scanResults.listen((results) {}).onData(_onPeripheralsDetectedData);
+  }
+
+  void _onPeripheralsDetectedData(List<ScanResult> results) {
+    for (final scanResult in results) {
+      /// Givt beacons have a name, manufacturer data, service data,
+      /// and a service UUID.
+      if (scanResult.device.name.isEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.manufacturerData.isNotEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.serviceData.isEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.serviceUuids.isEmpty) {
+        continue;
+      }
+      if (!scanResult.advertisementData.serviceData
+          .containsKey(scanResult.advertisementData.serviceUuids.first)) {
+        continue;
+      }
+
+      if (scanResult.rssi < -75 || scanResult.rssi < -67) {
```

Comment:
same here with values -75 and -67, what do they mean? Shall it be constants with clear names instead?

Reply:
[rssi](https://bluetoothle.wiki/rssi) is the power of the frequency of the bluetooth device. If the device is further away meaning that is -57 or -40 for example we are skipping it.